### DISCUSSION TITLE
Ranged background colors

### DISCUSSION THREAD

**Discussion by arijit4** [patrykandpatrick/vico#404] [Ideas]
Really great library... Everything worksgreat and smooth.

It would really be great if you could add range based background colors. I want to do something like this:
![Screenshot_20230923_154435_Chrome.jpg](https://github.com/patrykandpatrick/vico/assets/72606854/f4379357-5650-496b-9e4a-a616b5587341)
You can find it [here](https://codeforces.com/profile/tourist)

**Comment by patrickmichalik** [comment]
Hello! Thanks for the kind words. You can implement this as a [`Decoration`](https://patrykandpatrick.com/vico/api/core/com.patrykandpatrick.vico.core.chart.decoration/-decoration). I suggest overriding [`Decoration#onDrawBehindChart`](https://patrykandpatrick.com/vico/api/core/com.patrykandpatrick.vico.core.chart.decoration/-decoration/on-draw-behind-chart) and using a [`ShapeComponent`](https://patrykandpatrick.com/vico/api/core/com.patrykandpatrick.vico.core.component.shape/-shape-component) for each stripe.
